modules/apm/main.tf

```diff
@@ -0,0 +1,37 @@
+resource "newrelic_alert_policy" "main_policy" {
+  name = var.policy_name
+}
+
+resource "newrelic_nrql_alert_condition" "app_apdex_score_alert" {
+  count       = var.app_apdex_score_enabled == true ? 1 : 0
+  name        = var.app_apdex_score_message
+  description = var.app_apdex_score_message
+  runbook_url = var.app_apdex_score_runbook_url
+
+  type                 = "static"
+  policy_id            = newrelic_alert_policy.main_policy.id
+  violation_time_limit = "one_hour"
+  value_function       = "single_value"
+
+  nrql {
+    query             = <<-EOQ
+      SELECT apdex(duration, t:${var.app_apdex_score_timeframe}) FROM Transaction WHERE appName like ${local.appname_like}
```

Comment:
It's not a timeframe but a threshold:

> `apdex(attribute, t: )`
> Use the `apdex` function to return an Apdex score for a single transaction or for all your transactions. The `attribute` can be any attribute based on response time, such as `duration` or `backendDuration`. The `t:` argument defines an Apdex T *threshold* in seconds.

See https://docs.newrelic.com/docs/query-your-data/nrql-new-relic-query-language/get-started/nrql-syntax-clauses-functions#func-apdex

Reply:
Should be better now :)